src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/profiler/ProfilerEditingTarget.java

```diff
@@ -671,16 +696,61 @@ public void execute()
       };
    }
    
-   private static void onMessage(String message, String file, int line)
-   {  
+   private void onMessage(final String message,
+                          final String file,
+                          final String details,
+                          final int line)
+   {
       if (message == "sourcefile")
       {
-         FilePosition filePosition = FilePosition.create(line, 0);
-         CodeNavigationTarget navigationTarget = new CodeNavigationTarget(file, filePosition);
-         
-         RStudioGinjector.INSTANCE.getFileTypeRegistry().editFile(
-               FileSystemItem.createFile(navigationTarget.getFile()),
-               filePosition);
+         server_.profileSources(file, new ServerRequestCallback<String>()
```

Comment:
Ah, good to know! So I have plans to use `profile_sources` to fix some of the source paths; this is why the method returns a path not a boolean. However, I'm not yet convinced how much effort we want to put here since it's potentially and endless source of issues. If you don't mind, it would help me keep the code as is but I can definitely consider removing this later on if this remains unchanged.